src/EditorFeatures/Core/SettingsData/Analyzers/AnalyzerSettingInfo.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.Editor.SettingsData
+{
+    internal class AnalyzerSettingInfo
+    {
+        private readonly DiagnosticDescriptor _descriptor;
+        private readonly AnalyzerSettingsUpdaterService _settingsUpdateerService;
+        private readonly Action<string> _onSelectionChanged;
+
+        public AnalyzerSettingInfo(DiagnosticDescriptor descriptor,
+                                   ReportDiagnostic effectiveSeverity,
+                                   AnalyzerSettingsUpdaterService settingsUpdaterService,
+                                   Action<string> onSelectionChanged)
+        {
+            _descriptor = descriptor;
+            _settingsUpdateerService = settingsUpdaterService;
+            _onSelectionChanged = onSelectionChanged;
+            DiagnosticSeverity severity = default;
+            if (effectiveSeverity == ReportDiagnostic.Default)
+            {
+                severity = descriptor.DefaultSeverity;
+            }
+            else if (effectiveSeverity.ToDiagnosticSeverity() is DiagnosticSeverity severity1)
+            {
+                severity = severity1;
+            }
+
+            var enabled = effectiveSeverity != ReportDiagnostic.Suppress;
+            IsEnabled = enabled;
+            Severity = severity;
+        }
+
+        public string Id => _descriptor.Id;
+        public string Title => _descriptor.Title.ToString();
+        public string Description => _descriptor.Description.ToString();
+        public string Category => _descriptor.Category;
+        public DiagnosticSeverity Severity { get; private set; }
+        public bool IsEnabled { get; private set; }
+        private bool IsSelected { get; set; } = false;
+
+        internal void SetSelected()
+        {
+            if (IsSelected == false)
+            {
+                _onSelectionChanged?.Invoke(Description);
+                IsSelected = true;
+            }
+            else
+            {
+                IsSelected = false;
+            }
```

Comment:
`_onSelectionChanged` is only invoked when changing from false to true.

Looking at the name, I'd expect it to invoked for any change (true to false or false to true)